server/server.go

```diff
@@ -392,15 +394,27 @@ func (s *Server) handleReq(conn *gokeyless.Conn, ch chan *gokeyless.Packet) {
 }
 
 // Serve accepts incoming connections on the Listener l, creating a new service goroutine for each.
-func (s *Server) Serve(l net.Listener, timeout time.Duration) error {
+func (s *Server) Serve(l net.Listener) error {
 	defer l.Close()
+
+	ch := make(chan req, 8)
+	for i := 0; i < 8; i++ {
+		go s.handleReqs(ch)
+	}
+	defer func() {
+		s.workerMu.Lock()
+		s.closed = true
+		close(ch)
+		s.workerMu.Unlock()
+	}()
+
 	for {
 		c, err := l.Accept()
 		if err != nil {
 			log.Error(err)
 			return err
 		}
-		go s.handle(gokeyless.NewConn(tls.Server(c, s.Config)), timeout)
+		go s.handle(gokeyless.NewConn(tls.Server(c, s.Config)), ch)
```

Comment:
The goroutine will stop blocking and exit if the connection closes. Plus, for each connection, we only spawn a new goroutine, and goroutines are much cheaper than threads.

I did an experiment and made 5k simultaneous connections to a keyserver -- without this PR the server crashes, and with this PR memory stabilizes at 70mb.